server/cloudformation/ecs.go

```diff
@@ -120,6 +120,12 @@ func (p *ECSServiceResource) Provision(ctx context.Context, req customresources.
 			} else {
 				err = fmt.Errorf("no primary deployment found")
 			}
+		} else {
```

Comment:
We should probably just reverse the flow here so we check the error case first, and remove the duplication in getting the deployment id:

```go
if err != nil {
        if err, ok := err.(awserr.Error); ok && strings.Contains(err.Message(), "TaskDefinition is inactive") {
                // If the task definition has been previously deleted, just do
                // nothing.
        } else {
                return id, data, err
        }
}


d := primaryDeployment(resp.Service)
if d != nil {
        data["DeploymentId"] = *d.Id
} else {
        err = fmt.Errorf("no primary deployment found")
}
```